PowerEditor/src/WinControls/ProjectPanel/ProjectPanel.cpp

```diff
@@ -454,6 +455,9 @@ bool ProjectPanel::writeWorkSpace(TCHAR *projectFileName)
     if (!tvRoot)
       return false;
 
+	TCHAR * fileName = PathFindFileName(projectFileName);
+	_treeView.renameItem(tvRoot, fileName);
+

```

Comment:
1. Wrong indentation characters used.
2. Change to `TCHAR * fileName = PathFindFileName(fn2write);` (see first two lines of method).

Reply:
For filename: Thank you again, this one would crash after the _...workspace was modified. Do you want to save it?_ request. I should be more careful.

For indentation: In most of the code lines in this project, tabs are used for indentation.  In this function, however, in 18 lines spaces are used, and in 9 lines tabs. I added two lines with tabs, because I thought to make it equal to the majority of code lines. This has the unfortunate effect of the wrong indentation in this preview. In my IDE, it looks fine. But I can change to spaces here too.

Question remaining: spaces or tabs?

Reply:
>Question remaining: spaces or tabs?
> I thought to make it equal to the majority of code lines.

I think you did the right thing.

>wrong indentation in this preview. In my IDE, it looks fine.

Yes, this happens sometimes.

Reply:
@MAPJe71 

Since the PR made it into the master without the modifications you recommended, I checked if the crash I was afraid of would really happen. It does not. `PathFindFileName` returns `NULL`, if `NULL` is given as parameter, and `treeView.renameItem(tvRoot, fileName)` returns immediately without action, if `NULL` is specified as `fileName`.

The function `ProjectPanel::writeWorkSpace` is only called with `NULL` as parameter before Npp ends or before a new workspace is loaded, so that luckily not even a wrong display name is left behind.

Thank you again for watching, such things can go wrong.